### PR TITLE
Add shape function and dynamic shape checking for mlas_matmul

### PR DESCRIPTION
**Pull request by ymwangg**
Add shape function for mlas_matmul.

### CODE REVIEW

**Review by dvhg — APPROVED** [review]

File: python/tvm/topi/x86/dense_alter_op.py [hunk 1]
```diff
@@ -36,6 +36,11 @@ def _alter_dense_layout(attrs, inputs, tinfos, out_type):
         and tinfos[1].dtype == "float32"
         and out_type.dtype == "float32"
     ):
+        # mlas is only used for static tensors
+        if any([isinstance(dim, tvm.tir.Any) for dim in tinfos[0].shape]) or any(
+            [isinstance(dim, tvm.tir.Any) for dim in tinfos[1].shape]
+        ):
+            return None
```
Comment (trevor-m) [inline python/tvm/topi/x86/dense_alter_op.py]: I think we want to continue to non-mlas code after this instead of returning none.

**Review by trevor-m — CHANGES_REQUESTED** [review]

File: python/tvm/topi/x86/dense_alter_op.py (hunk 1 above)
Comment (ymwangg) [reply python/tvm/topi/x86/dense_alter_op.py]: Thanks, good catch!

Comment (ymwangg) [reply python/tvm/topi/x86/dense_alter_op.py]: Updated.

**Review by trevor-m — APPROVED** [review]
Thanks @ymwangg!

**Pull request merged**